PTGLgraphComputation/src/proteingraphs/ComplexGraph.java

```diff
@@ -265,15 +265,22 @@ private void createVertices(List<Chain> chains) {
                         proteinNodeMap.put(v, tmpChain.getPdbChainID());
                         labelNodeMap.put(v, labelList.length > curNum ? labelList[curNum] : tmpChain.getPdbChainID());
                         molMap.put(v, FileParser.getMetaInfo(pdbid, tmpChain.getPdbChainID()).getMolName());  // get the mol name from the ProtMetaInfo
-                        chainLengthMap.put(v, tmpChain.getAllAAResidues().size());
+                        if(includeRna && allChains.get(j).getMoleculeType().contains("polyribonucleotide")){
+                            chainLengthMap.put(v, tmpChain.getAllAAResidues().size() + tmpChain.getAllRnaResidues().size());
+                        }else{
+                            chainLengthMap.put(v, tmpChain.getAllAAResidues().size());
+                        }
                         chainTypeMap.put(v, allChains.get(j).getMoleculeType());
                         mapVertexIdVertex.put(Integer.parseInt(v.toString()), v);
 
                         molIDs.add(FileParser.getMetaInfo(pdbid, tmpChain.getPdbChainID()).getMolName());
                         
-                        int chainLength = (Settings.getBoolean("PTGLgraphComputation_B_CG_ignore_ligands") ? 
-                                tmpChain.getAllAAResidues().size() : 
-                                tmpChain.getAllAAResidues().size() + tmpChain.getAllLigandResidues().size());
+                        int chainLength = (Settings.getBoolean("PTGLgraphComputation_B_CG_ignore_ligands") ?    // added extra case becuase RNA is not inculded in allAAresidues anymore
+                                ((includeRna) ? tmpChain.getAllAAResidues().size() + tmpChain.getAllRnaResidues().size() :
+                                        tmpChain.getAllAAResidues().size()) : ((includeRna) ?
+                                tmpChain.getAllAAResidues().size() + tmpChain.getAllRnaResidues().size() +
+                                        tmpChain.getAllLigandResidues().size() : tmpChain.getAllAAResidues().size() +
+                                tmpChain.getAllLigandResidues().size()));
```

Comment:
Please transform to if-else-statements instead of using nested ternary operator for better clarity. Maybe something like this would be easiest to understand:
```
int tmpLength = all AA residues
if not ignore ligands:
  tmpLength += all ligands
if includeRNA
  tmpLength += all RNA
```